src/Serilog.Sinks.Couchbase/Sinks/Couchbase/CouchbaseSink.cs

```diff
@@ -99,15 +136,27 @@ protected override void EmitBatch(IEnumerable<Events.LogEvent> events)
             // This sink doesn't actually write batches, instead only using
             // the PeriodicBatching infrastructure to manage background work.
             // Probably needs modification.
+            IBucket bucket = null;
 
-            foreach (var logEvent in events)
+            try
             {
-                var key = Guid.NewGuid().ToString();
-                
-                var result = _couchbaseClient.StoreJson(Enyim.Caching.Memcached.StoreMode.Add, key, new LogEvent(logEvent, logEvent.RenderMessage(_formatProvider)));
+                bucket = _couchbaseCluster.OpenBucket(_bucketName, _bucketPassword);
 
-                if (!result)
+                foreach (var result in events.Select(logEvent => bucket.Insert(new Document<LogEvent>
```

Comment:
Writing this as

```csharp
foreach (var evt in events)
{
    if (!bucket.Insert(...))
        SelfLog.WriteLine(...);
}
```

would keep the code further left and make debugging with breakpoints a bit easier.